jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarModeProcessor.java

```diff
@@ -131,6 +138,81 @@ static JarType determineJarType(Path jarPath) throws IOException {
     return layers;
   }
 
+  /**
+   * Creates layers as specified in BOOT-INF/layers.idx (if present) or for dependencies,
+   * spring-boot-loader, snapshot dependencies, resource and classes for a spring boot fat jar.
+   *
+   * @param jarPath path to jar file
+   * @param localExplodedJarRoot path to temporary jib local directory
+   * @return list of {@link FileEntriesLayer}
+   * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
+   *     provided doesn't exist
+   */
+  static List<FileEntriesLayer> createLayersForExplodedSpringBootFat(
+      Path jarPath, Path localExplodedJarRoot) throws IOException {
+
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      ZipEntry layerIndex = jarFile.getEntry("BOOT-INF/layers.idx");
+      ZipUtil.unzip(jarPath, localExplodedJarRoot);
+      if (layerIndex != null) {
+        return createLayersForLayeredSpringBootJar(localExplodedJarRoot);
+      }
+
+      ArrayList<FileEntriesLayer> layers = new ArrayList<>();
+
+      // Non-snapshot layer
+      Predicate<Path> isInBootInfLib =
+          path -> path.getParent().startsWith(localExplodedJarRoot.resolve("BOOT-INF/lib"));
```

Comment:
Although `BOOT-INF/lib` works, for non-test code, we do `.resolve("BOOT-INF").resolve("lib")` to avoid using a certain path separator.

And why do we need `getParent()` if we are checking `startsWith()`?

Reply:
Oops, my bad.